### PR TITLE
riscv: build and test on CI

### PR DESCRIPTION
**Pull request by 64**

### CODE REVIEW

**Review by avdgrinten — COMMENTED** [review]
This PR duplicates a lot of code in `ci.yml`. Can't we avoid that by using a build matrix? That would also make it easier to extend the CI to aarch64.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ jobs:
     build-mlibc:
         strategy:
             matrix:
+                arch: [x86_64, riscv64]
```
Comment (Geertiebear) [inline .github/workflows/ci.yml:9]: Why not aarch64 too?

Comment (64) [reply .github/workflows/ci.yml:9]: We don't have Linux sysdeps for AArch64 yet

**Comment by 64** [comment]
Looks like this is ready again.

File: sysdeps/linux/generic/sysdeps.cpp [hunk 2]
```diff
@@ -359,6 +359,11 @@ int sys_clone(void *entry, void *user_arg, void *tcb, pid_t *pid_out) {
 extern "C" const char __mlibc_syscall_begin[1];
 extern "C" const char __mlibc_syscall_end[1];
 
+#if defined(__riscv)
+// Disable UBSan here to work around qemu-user misaligning ucontext_t.
+// https://github.com/qemu/qemu/blob/2bf40d0841b942e7ba12953d515e62a436f0af84/linux-user/riscv/signal.c#L68-L69
+[[gnu::no_sanitize("undefined")]]
```
[the remark this answers is not in the document]
Comment (avdgrinten) [reply sysdeps/linux/generic/sysdeps.cpp:365]: The qemu comment points to the kernel and indeed, [it seems that the kernel misaligns the struct](https://github.com/torvalds/linux/blob/master/arch/riscv/kernel/signal.c#L170). This means that we should probably override the struct's alignment using an attribute.

[the remark this answers is not in the document]
Comment (64) [reply sysdeps/linux/generic/sysdeps.cpp:365]: A warning seems a little ugly here. Maybe a better fix would be to make the struct have alignment 1 internally (but keep it aligned to 16 in the headers for libc consumers).

**Review by Geertiebear — APPROVED** [review]
LGTM in general.

**Pull request merged**